backend/shelf-server/src/main/java/com/htec/shelfserver/service/UserService.java

```diff
@@ -183,8 +187,51 @@ public UserResponseModel updateUserRole(Long id, Long roleId) {
         model.put("firstName", user.getFirstName());
         model.put("roleName", user.getRole().getName());
 
-        emailService.sendEmail(user.getEmail(), model, "update-role-email.html", "Role updated" );
+        emailService.sendEmail(user.getEmail(), model, "update-role-email.html", "Role updated");
 
         return UserMapper.INSTANCE.userEntityToUserResponseModel(user);
     }
+
+    public void resetPassword(PasswordResetModel passwordResetModel) {
+
+        String password = passwordResetModel.getPassword();
+        userValidator.isUserPasswordValid(password);
+        String jwtToken = passwordResetModel.getJwtToken();
+
+        if(!checkUserByJwtToken(jwtToken))
+            throw ExceptionSupplier.tokenNotValid.get();
+
+        if (!passwordResetTokenRepository.findByToken(jwtToken).isPresent())
+            throw ExceptionSupplier.emailResetRequestWasNotSent.get();
+
+        long userId = findUserIdByJwtToken(jwtToken);
+        UserEntity userEntity = userRepository.findById(userId).orElseThrow(ExceptionSupplier.recordNotFoundWithEmail);
+        String salt = userEntity.getSalt();
+        String newPassword = bCryptPasswordEncoder.encode(password + salt);
+        userEntity.setPassword(newPassword);
+        userRepository.save(userEntity);
+        passwordResetTokenRepository.deleteAllByUserDetails(userEntity);
+    }
+
+    public boolean checkUserByJwtToken(String jwtToken) {
+
+        String user;
+        try {
+            user = Jwts.parser()
+                    .setSigningKey(SecurityConstants.TOKEN_SECRET)
+                    .parseClaimsJws(jwtToken)
+                    .getBody()
+                    .getSubject();
+        } catch (ExpiredJwtException e) {
+            throw ExceptionSupplier.tokenExpired.get();
+        } catch (Exception e) {
+            throw ExceptionSupplier.tokenNotValid.get();
+        }
+
+        return user != null;
+    }
+
+    public long findUserIdByJwtToken(String jwtToken) {
+        return passwordResetTokenRepository.findByToken(jwtToken).get().getUserDetails().getId();
```

Comment:
Potential nullpointer

Reply:
findUserIdByJwtToken should never throw nullpointer because on line 204 there is an if which checks if there is an entry in with jwt token in db which throws exception.